ui/components/app/assets/hooks/useMultichainAssets.tsx

```diff
@@ -0,0 +1,75 @@
+import { useSelector } from 'react-redux';
+import { CaipAssetId, Hex } from '@metamask/utils';
+import { getMultichainSelectedAccountCachedBalance } from '../../../../selectors/multichain';
+import { getSelectedInternalAccount } from '../../../../selectors';
+import {
+  TranslateFunction,
+  networkTitleOverrides,
+} from '../util/networkTitleOverrides';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import { formatWithThreshold } from '../util/formatWithThreshold';
+import { getIntlLocale } from '../../../../ducks/locale/locale';
+import { getCurrentCurrency } from '../../../../ducks/metamask/metamask';
+import {
+  MULTICHAIN_PROVIDER_CONFIGS,
+  MultichainNetworks,
+} from '../../../../../shared/constants/multichain/networks';
+import { TokenWithFiatAmount } from '../types';
+import { getMultiChainAssets } from '../../../../selectors/assets';
+
+const useMultiChainAssets = () => {
+  const t = useI18nContext();
+  const locale = useSelector(getIntlLocale);
+  const selectedAccount = useSelector(getSelectedInternalAccount);
+  const currentCurrency = useSelector(getCurrentCurrency);
+
+  const multichainAssets = useSelector((state) =>
+    getMultiChainAssets(state, selectedAccount),
+  );
+
+  // the following condition is needed to satisfy e2e check-balance.spec.ts
+  // this is because the new multichain data is not being mocked within the withSolanaAccountSnap test fixture
+  // balances render as expected without this condition during local testing
+  const cachedBalance = useSelector(getMultichainSelectedAccountCachedBalance);
+  if (cachedBalance === 0) {
+    return [
+      {
+        chainId: MultichainNetworks.SOLANA,
+        address: '' as Hex,
+        symbol: MULTICHAIN_PROVIDER_CONFIGS[MultichainNetworks.SOLANA].ticker,
+        string: `${cachedBalance} ${currentCurrency}`,
+        primary: cachedBalance,
+        image: '',
+        secondary: cachedBalance,
+        tokenFiatAmount: cachedBalance,
+        isNative: true,
+        decimals: 9, // hard coded decimal value
+        title: MULTICHAIN_PROVIDER_CONFIGS[MultichainNetworks.SOLANA].nickname,
+        isStakeable: false,
+      },
+    ];
+  }
+
+  return multichainAssets.map((asset: TokenWithFiatAmount) => {
+    const assetId = asset.address as CaipAssetId;
+    const assetDetails = assetId.split('/')[1];
+    const isToken = assetDetails.split(':')[0] === 'token';
+
+    const fiatAmount = formatWithThreshold(asset.secondary, 0.01, locale, {
+      style: 'currency',
+      currency: currentCurrency.toUpperCase(),
+    });
+
+    return {
+      ...asset,
+      title: isToken
+        ? asset.title
+        : networkTitleOverrides(t as TranslateFunction, {
+            title: asset.title,
+          }),
+      secondary: fiatAmount, // secondary balance (usually in fiat)
+    };
```

Comment:
Rather than parsing the token id again can't we use the `isNative` property directly ? it's doing the same thing.

Reply:
I think that should work too, the parsing here seems to be done to distinguish native from non-native tokens which can be achieved by relying on the isNative property; can update this, no need to refactor and use the utils fct from here https://github.com/MetaMask/utils/blob/cba07ff6fd0fe57b3a41063fbf81f990a09b3590/src/caip-types.ts#L309 in that case;

Reply:
You will need it since this parsing is also done in `assets.ts` selector file :)